### ISSUE TITLE
Support non-UTF-8 character in binary_data

### ISSUE THREAD

**Issue by pixelshot91** [apache/datafusion-sqlparser-rs#1300]
`Parser::try_from_sql` and `Parser::parse_sql` both take a `&str` as input.
But `str` only allow for UTF-8 character
Is there any function to parse SQL statement from binary data (eg `Vec<u8>`) ?
Is there any plan to support it ?

**Comment by alamb** [comment]
I don't think we have any plans to support this as I don't know of any database that supports it

Perhaps you could use one of the binary string representations that escape binary data

Something like this in postgres

```sql
U&"d\0061t\+000061"
```
